### PR TITLE
Automatically fetch book covers from google books.

### LINKED ISSUE #131: Covers
Somewhat related to #106 (for search results).

Options: 
 * Umlaut (would require an OpenURL embedded on the page cf. #85)
 * Hit Amazon, Google, Hathi directly

### PR DESCRIPTION
**Pull request by terrellt**
Closes #131

### CODE REVIEW

File: app/assets/javascripts/book_covers.js.coffee [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+class BookCoverManager
+  constructor: ->
+    this.find_book_covers()
+  find_book_covers: ->
+    number = this.isbn() || this.oclc()
+    return unless number?
+    url = "#{@google_url}#{number}"
+    $.getJSON(url, this.process_record)
+  isbn: ->
+    isbn = $("dd.blacklight-isbn_display li")
```
Comment (tampakis) [inline app/assets/javascripts/book_covers.js.coffee]: There is a normalized version of the ISBN in the [isbn_s](https://github.com/pulibrary/orangeindex/blob/development/lib/traject_config.rb#L727) field.

**Comment by tampakis** [comment]
The Google Book api also accepts LCCN as an identifier. Could pass that over in addition to ISBN/OCLC? The field is [lccn_s](https://github.com/pulibrary/orangeindex/blob/development/lib/traject_config.rb#L715).

File: app/assets/javascripts/book_covers.js.coffee (hunk 1 above)
Comment (terrellt) [inline app/assets/javascripts/book_covers.js.coffee]: Yeah, but I don't think this is in the view, right?

Comment (tampakis) [inline app/assets/javascripts/book_covers.js.coffee]: Good point. Would it make sense to include it without displaying it?

Comment (terrellt) [inline app/assets/javascripts/book_covers.js.coffee]: I could do that - is there a concern that the ISBN in the display isn't queryable?

**Comment by terrellt** [comment]
> The Google Book api also accepts LCCN as an identifier. Could pass that over in addition to ISBN/OCLC? The field is lccn_s.

Done.

**Comment by jpstroop** [comment]
FWIW, lccn is likely to get us the most of the three, so if there's a preferred order, I'd go w/ that first.

**Comment by terrellt** [comment]
@jpstroop It'll try them all and stop when it finds a cover. In the test data it was only ever isbn that returned one.

**Comment by jpstroop** [comment]
> if there's a preferred order

_and_ it makes a difference.

**Comment by jpstroop** [comment]
:+1:

**Comment by tampakis** [comment]
:+1:

**Pull request merged**